### PR TITLE
PA-936 Settings up basic prometheus and grafana in the tools project

### PR DESCRIPTION
**Pull request by goemen**

### CODE REVIEW

**Comment by goemen** [comment]
I will need some Github OAuth data to finalize Grafana deployment @Fosol

**Review by jdtoombs — APPROVED** [review]

File: openshift/templates/monitoring/grafana-template.yml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        name: pims-monitoring-tools-grafana-ini-config
+        app: pims
+        component: monitoring
+        env: tools
+        role: prometheus
+    type: Opaque
+    data:
+      grafana.ini: |-
+        [server]
+        root_url = https://grafana-jcxjin-tools.pathfinder.gov.bc.ca/
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:50]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        domain = grafana-jcxjin-tools.pathfinder.gov.bc.ca
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:51]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        - name: Prometheus
+          type: prometheus
+          access: proxy
+          url: https://prometheus-jcxjin-tools.pathfinder.gov.bc.ca
+
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: pims-monitoring-tools-grafana-ini-config
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:37]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        role: prometheus
+    type: Opaque
+    data:
+      datasources.yml: |-
+        apiVersion: 1
+        datasources:
+        - name: Prometheus
+          type: prometheus
+          access: proxy
+          url: https://prometheus-jcxjin-tools.pathfinder.gov.bc.ca
```
[the remark this answers is not in the document]
Comment (Fosol) [reply openshift/templates/monitoring/grafana-template.yml:31]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      namespace: jcxjin-tools
+  - kind: Route
+    apiVersion: v1
+    metadata:
+      labels:
+        name: grafana
+      name: grafana
+      namespace: jcxjin-tools
+    spec:
+      host: "grafana-jcxjin-tools.pathfinder.gov.bc.ca"
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:79]: Should probably be a variable
I wonder if we should rename to something more generic `pims-grafana.pathfinder.gov.bc.ca`?

File: openshift/templates/monitoring/grafana-template.yml [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        kind: Service
+        weight: 100
+      tls:
+        termination: edge
+
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: grafana
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:91]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+      labels:
+        name: ${GRAFANA_SERVICE_ACCOUNT_NAME}
+      namespace: jcxjin-tools
+  - kind: Route
+    apiVersion: v1
+    metadata:
+      labels:
+        name: grafana
+      name: grafana
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:77]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+          protocol: TCP
+          targetPort: 3000
+      selector:
+        app: pims-grafana-tools
+
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: pims-grafana-password-secret
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:108]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+        role: grafana
+    type: Opaque
+    stringData:
+      GF_SECURITY_ADMIN_PASSWORD: ${GF_SECURITY_ADMIN_PASSWORD}
+
+  - kind: DeploymentConfig
+    apiVersion: v1
+    metadata:
+      name: pims-grafana-tools
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:124]: Should probably be a variable

File: openshift/templates/monitoring/grafana-template.yml [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+            - name: "pims-monitoring-grafana-ini-config-volume"
+              configMap:
+                name: pims-monitoring-tools-grafana-ini-config
+          containers:
+            - name: pims-grafana-tools
+              image: grafana/grafana
+              imagePullPolicy: Always
+              env:
+                - name: GF_SERVER_ROOT_URL
+                  value: https://grafana-jcxjin-tools.pathfinder.gov.bc.ca
```
Comment (Fosol) [inline openshift/templates/monitoring/grafana-template.yml:172]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+    description: Deployment template for Prometheus
+    tags: pims-monitoring
+objects:
+  - kind: "PersistentVolumeClaim"
+    apiVersion: v1
+    metadata:
+      name: "prometheus-service-volume-claim"
+      labels:
+        name: prometheus-service-volume-claim
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:16]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+      resources:
+        requests:
+          storage: 5Gi
+    volumeName: "prometheus-service-volume-claim"
+
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: pims-prometheus-tools-config
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:30]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+  - kind: Route
+    apiVersion: v1
+    metadata:
+      labels:
+        name: prometheus
+      name: prometheus
+      namespace: jcxjin-tools
+    spec:
+      host: "prometheus-jcxjin-tools.pathfinder.gov.bc.ca"
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:72]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+          static_configs:
+          - targets: 
+            - ${PIMS_PROD_API_URL}
+  - kind: ServiceAccount
+    apiVersion: v1
+    metadata:
+      name: ${PROMETHEUS_SERVICE_ACCOUNT_NAME}
+      labels:
+        name: ${PROMETHEUS_SERVICE_ACCOUNT_NAME}
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:63]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+  - kind: Route
+    apiVersion: v1
+    metadata:
+      labels:
+        name: prometheus
+      name: prometheus
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:70]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+        termination: edge
+
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: prometheus
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/scheme: http
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:87]: Should probably be a variable

File: openshift/templates/monitoring/prometheus-template.yml [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+          protocol: TCP
+          targetPort: 9090
+      selector:
+        app: pims-monitoring-tools
+
+  - kind: DeploymentConfig
+    apiVersion: v1
+    metadata:
+      name: pims-monitoring-tools
+      namespace: jcxjin-tools
```
Comment (Fosol) [inline openshift/templates/monitoring/prometheus-template.yml:104]: Should probably be a variable

**Review by Fosol — APPROVED** [review]

**Pull request merged**